MongoRepository/Repository/MongoRepository.cs

```diff
@@ -146,6 +167,21 @@ public virtual T Update(T entity)
         }
 
         /// <summary>
+        /// Upserts an entity asynchronously.
+        /// </summary>
+        /// <param name="entity">The entity.</param>
+        /// <returns>The updated entity.</returns>
+        public virtual async Task<T> UpdateAsync(T entity)
+        {
+            if (entity.Id == null)
+                await this.AddAsync(entity);
```

Comment:
Isn't possible to use UpdateOneAsync here and sending Update options upsert = true. I've not looked into the new API but can't we build UpdateDefintion from entity (which means update everything) as default?